pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -687,16 +681,14 @@ func (c *InstallerController) newNodeStateForInstallInProgress(ctx context.Conte
 			return ret, false, fmt.Sprintf("operand pod failed: %v", strings.Join(ret.LastFailedRevisionErrors, "\n")), nil
 
 		case staticPodStateReady:
-			ret := currNodeState.DeepCopy()
-			if currNodeState.TargetRevision > ret.CurrentRevision {
-				ret.CurrentRevision = currNodeState.TargetRevision
+			rev := currNodeState.CurrentRevision
+			if currNodeState.TargetRevision > currNodeState.CurrentRevision {
+				rev = currNodeState.TargetRevision
 			}
-			ret.TargetRevision = 0
-			ret.LastFailedRevision = 0
-			ret.LastFailedTime = nil
-			ret.LastFailedCount = 0
-			ret.LastFailedRevisionErrors = nil
-			return ret, false, staticPodReason, nil
+			return &operatorv1.NodeStatus{
+				NodeName:        currNodeState.NodeName,
+				CurrentRevision: rev,
+			}, false, staticPodReason, nil
```

Comment:
@p0lyn0mial this is the code that will be done by the startup-monitor in SNO (next PR).